### ISSUE TITLE
LayerDrop multiple GPU command

### ISSUE THREAD

**Issue by sshleifer** [pytorch/fairseq#2389]
I followed the layerdrop docs from @huihuifan on 8 GPUS for to replicate `layerdrop_wmt_en_de_12_6` and ended up with the following command:

```bash
fairseq-train \
    data-bin/wmt16_en_de_bpe32k \
    --arch transformer_vaswani_wmt_en_de_big --share-all-embeddings \
    --optimizer adam --adam-betas '(0.9, 0.98)' --clip-norm 0.0 \
    --lr 0.0005 --lr-scheduler inverse_sqrt --warmup-updates 4000 --warmup-init-lr 1e-07 \
    --dropout 0.3 --weight-decay 0.0 \
    --criterion label_smoothed_cross_entropy --label-smoothing 0.1 \
    --max-tokens 3584 \
    --fp16 \
    --encoder-layerdrop 0.2 --decoder-layerdrop 0.2 \
    --distributed-world-size 8 | tee fairseq_logs_layerdrop.txt
```
This produced an error, suggesting that "some parameters aren't contributing to the loss, use `--find-unused-parameters`, so I did that and now the command is running well.

I'm wondering whether adding `--find-unused-parameters` is expected/ok, or if it might cause unindented side effects, like the model being pruned permanently after step 0.
Thanks in advance!

Env:

- torch 1.5.1
- fairseq master 108bb256 (Jul 28 AM)
- cuda 10.1
- 8 v100 GPUS
- apex installed from source

**Comment by sshleifer** [comment]
Here are the metrics after 3 epochs:
```bash
2020-07-28 13:52:03 | INFO | valid | epoch 001 | valid on 'valid' subset | loss 11.758 | nll_loss 11.098 | ppl 2191.47 | wps 341069 | wpb 16782.6 | bsz 600 | num_updates 5462
2020-07-28 13:52:12 | INFO | train | epoch 001 | loss 11.125 | nll_loss 10.465 | ppl 1413.16 | wps 197521 | ups 7.85 | wpb 25150.9 | bsz 822.5 | num_updates 5462 | lr 0.000427882 | gnorm 1.858 | loss_scale 4 | train_wall 674 | wall 735
2020-07-28 14:03:36 | INFO | valid | epoch 002 | valid on 'valid' subset | loss 12.235 | nll_loss 11.592 | ppl 3086.26 | wps 292097 | wpb 16782.6 | bsz 600 | num_updates 10931 | best_loss 11.758
2020-07-28 14:04:02 | INFO | train | epoch 002 | loss 10.919 | nll_loss 10.233 | ppl 1203.28 | wps 193520 | ups 7.69 | wpb 25151.1 | bsz 822.5 | num_updates 10931 | lr 0.000302461 | gnorm 5.111 | loss_scale 0 | train_wall 668 | wall 1446
2020-07-28 14:15:27 | INFO | valid | epoch 003 | valid on 'valid' subset | loss 12.16 |
 nll_loss 11.544 | ppl 2985.29 | wps 337505 | wpb 16782.6 | bsz 600 | num_updates 16400 | best_loss 11.758
2020-07-28 14:15:56 | INFO | train | epoch 003 | loss 10.634 | nll_loss 9.901 | ppl 955.76 | wps 192786 | ups 7.66 | wpb 25151.8 | bsz 822.6 | num_updates 16400 | lr 0.0002
46932 | gnorm 3.609 | loss_scale 0 | train_wall 668 | wall 2160

```

**Comment by sshleifer** [comment]
Adding `--lr 0.001 --update-freq 16` helps a lot.

Full command:

```bash
export save_dir=ckpt/layerdrop_v2
fairseq-train \
    data-bin/wmt16_en_de_bpe32k \
    --arch transformer_vaswani_wmt_en_de_big --share-all-embeddings \
    --optimizer adam --adam-betas '(0.9, 0.98)' --clip-norm 0.0 \
    --lr 0.001 --update-freq 16 --lr-scheduler inverse_sqrt --warmup-updates 4000 --warmup-init-lr 1e-07 \
    --dropout 0.3 --weight-decay 0.0 --task translation \
    --decoder-layerdrop 0.3 --encoder-layerdrop 0.3 --encoder-layers 12 --decoder-layers 6 \
    --criterion label_smoothed_cross_entropy --label-smoothing 0.1 \
    --max-tokens 3584 \
    --fp16 \
    --save-interval 1 --save-interval-updates 5000 --keep-interval-updates=-1  --tensorboard-logdir tboard \
    --distributed-world-size 8 --ddp-backend no_c10d --save-dir $save_dir | tee $save_dir/logs.txt
```

**Comment by songyouwei** [comment]
add `--ddp-backend no_c10d` works for me

**Comment by sunmeng01** [comment]
@sshleifer Hi, Can you get the bleu score 30.2 reported in the paper using the above script? After averaging 10 latest checkpoints,   the best bleu score I got is only 29.64

**Comment by sshleifer** [comment]
I don't remember the exact numbers I got, but I would have been satisfied with 29.64.

**Comment by sunmeng01** [comment]
Thanks
